source/grid/grid_in.cc

```diff
@@ -1926,14 +1980,30 @@ GridIn<dim, spacedim>::read_msh(std::istream &in)
                       vertex = numbers::invalid_unsigned_int;
                     }
               }
-            else if ((cell_type == 3) && (dim == 3))
+            else if ((cell_type == 2 || cell_type == 3) && (dim == 3))
```

Comment:
I know it is not your fault, but shouldn't this be triggered also for `cell_type == 1`, i.e., edges at the boundary?

Reply:
(This may be addressed in another PR though).